### DISCUSSION TITLE
Recommendation for skipping already-downloaded files

### DISCUSSION THREAD

**Discussion by MineSafety** [sec-edgar/sec-edgar#242] [Q&A]
Is there a recommended way to skip filings that have already been downloaded? I know I can probably use some combination of `os.path` + `get_urls` + `get_accession_number(url)`, but is there an argument that can be passed to `my_filings.save` which would essentially skip a download if the file is already in the directory?

**Comment by jackmoody11** [comment]
> Is there a recommended way to skip filings that have already been downloaded? I know I can probably use some combination of `os.path` + `get_urls` + `get_accession_number(url)`, but is there an argument that can be passed to `my_filings.save` which would essentially skip a download if the file is already in the directory? 

Thanks for the question @MineSafety. Right now, there isn't a great way to do this. I think it will require some refactoring of the save method and some of the methods used within that method.


On a related note, I would really like to simplify the code for saving filings, but am not sure how. If you have any suggestions, I'd love to hear them.

**Reply by gyund** [reply]
Not sure if something like [diskcache](https://grantjenks.com/docs/diskcache/api.html#diskcache.Cache.memoize) would help.
